File: src/main/kotlin/codeacademy/bookingforum/app/topicCategory/TopicCategoryController.java
package codeacademy.bookingforum.app.topicCategory;

import codeacademy.bookingforum.app.configuration.ResponseObject;
import jakarta.validation.Valid;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.security.access.annotation.Secured;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.context.request.WebRequest;

import java.util.List;

@RestController
@RequestMapping("/api/section")
public class TopicCategoryController {
    @Autowired
    TopicCategoryService sectionService;

    // Create new section (TopicCategory)
    @Secured("ROLE_ADMIN")
    @PostMapping("/create")
    @ResponseStatus(HttpStatus.CREATED)
    public ResponseObject create(@RequestBody @Valid TopicCategoryDto section, WebRequest request) {
        return sectionService.create(section, request);
    }

    // Get all section id's
    @GetMapping("/list")
    public List<TopicCategoryDto> getList() {
        return sectionService.getList();
    }
}

File: src/main/kotlin/codeacademy/bookingforum/app/topicCategory/TopicCategoryController.kt
package codeacademy.bookingforum.app.topicCategory

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/topicCategory")
class TopicCategoryController {

    @Autowired
    private lateinit var service: TopicCategoryService
    @GetMapping("/getAllTopicCategories")
    fun getTopicCategories(): List<TopicCategoryDto?> {
        return service.getAllTopicCategories()
    }

    @GetMapping("/getTopicCategory/{id}")
    fun getTopicCategoryById(@PathVariable id: Long): TopicCategoryDto? {
        return service.getAllTopicCategoryById(id)
    }

    @PostMapping("/postTopicCategory")
    fun postTopicCategory(@RequestBody topicCategoryDto: TopicCategoryDto?) {
        service.postTopicCategory(topicCategoryDto)
    }

    @PutMapping("/updateTopicCategory{id}")
    fun updateTopicCategory(@PathVariable id: Long, @RequestBody topicCategoryDto: TopicCategoryDto?){
        service.updateTopicCategory(id, topicCategoryDto)
    }

    @DeleteMapping("/deleteTopicCategory{id}")
    fun deleteTopicCategory(@PathVariable id: Long): String{
        return service.deleteTopicCategory(id)
    }
}